MailCore/Cache/MailboxManager/MailboxManager+Folders.swift

```diff
@@ -99,6 +99,23 @@ public extension MailboxManager {
         return frozenFolder
     }
 
+    func deleteFolder(folder: Folder) async throws {
+        try await apiFetcher.delete(mailbox: mailbox, folder: folder)
+        guard let liveFolder = folder.thaw() else { return }
+        try writeTransaction { writableRealm in
+            writableRealm.delete(liveFolder)
+        }
+    }
+
+    func modifyFolder(name: String, folder: Folder) async throws {
+        let modifiedFolder = try await apiFetcher.modify(mailbox: mailbox, folder: folder, name: name)
+        guard let liveFolder = folder.thaw() else { return }
+        try writeTransaction { writableRealm in
+            writableRealm.delete(liveFolder)
+            writableRealm.add(modifiedFolder, update: .modified)
+        }
```

Comment:
I looked it up. Trying to recompute locally would be too difficult (WS also doesn't do it)
```suggestion
                _ = try await apiFetcher.modify(mailbox: mailbox, folder: folder, name: name)
        try await refreshAllFolders()
```